### PR TITLE
Added macos-setup.md file and updated .gitignore

### LINKED ISSUE #157: Create New Developer Setup Guide
The current guide for setting up the developer's environment is unclear and hard to follow, especially for MacOS. Some of the steps were out of order, such as the steps for downloading the nltk library in the requirements.txt document, and there are some steps that had to be figured out independently. I suggest making a new markdown file for MacOS developers. The file would provide clear directions as to how to setup the environment to avoid any confusion or issues in the future. It would be beneficial by helping improve the onboarding experience.

- [x] Research and document the correct steps for setting up the environment on MacOS
- [x] Write a new markdown file with specific instructions as to how to set up the environment
- [x] Make sure to include any prerequisites and dependencies needed
- [x] Update the README.md file to reference the new developer setup guide

### PR DESCRIPTION
**Pull request by emrameduseljac**
Fixes #157 

**What was changed?**

* Added macos-setup.md file, which describes the steps for setting up MacOS developer environment.
* Added reference to the new markdown file in README.
* Removed the MacOS steps from the requirements.txt file.
* Updated the .gitignore file by adding venv/.

**Why was it changed?**

* The steps for MacOS setup were updated to be in order and listed everything that was missing and had to be downloaded.
* The prerequisites required for the environment were also mentioned to avoid any confusion.
* The steps in requirements.txt file for MacOS was out of order and did not work.
* The files were too large to be able to push so I had to add the virtual environment files to .gitignore.

**How was it changed?**

* When troubleshooting the issues with setting up the developer's environment, I kept a .txt file of all the commands I used. I used that Git history to figure out what steps I used to successfully set up my environment.
* I created the new markdown file by looking through my Git history, and I reorganized it into an order that made better sense/was easier to follow for the sake of the markdown file.
* I figured out I needed to add venv/ to the .gitignore file after attempting to push and getting an error that venv/ was too large to be pushed to the repo.

### CODE REVIEW

File: macos-setup.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+Ensure you have the following prerequisites downloaded:
+
+* `Java` --> [Download Here](https://www.oracle.com/java/technologies/downloads/)
+* `MySQL` --> [Download Here](https://dev.mysql.com/downloads/mysql/)
+
+Once those are downloaded and you have cloned the Git repository, you will want to go into the SpeechTranscription directory and create/activate a Python virtual environment:
+
+* `python3 -m venv venv` to create the environment.
+* `source venv/bin/activate` to activate.
+
```
Comment (emrameduseljac) [inline macos-setup.md:12]: Probably should mention that the venv will need to be activated everytime you want to run the program

File: README.md [hunk 2]
```diff
@@ -17,6 +17,8 @@ In this application, you can record live or upload an audio file form your compu
 # Contributing as a Developer
 
 ## Development Environment
+
+* If you are on MacOS, please refer to the macos-setup.md file.
```
Comment (emrameduseljac) [inline README.md:21]: Could include all operating systems on the new markdown file rather than only focusing it on MacOS

Comment (samfred) [reply README.md:21]: I agree as I have been having issues installing the requirements on Windows.

File: macos-setup.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        export SSL_CERT_FILE=${CERT_PATH}
+        export REQUESTS_CA_BUNDLE=${CERT_PATH}
+
+    3. Open Python Interpreter and import/download nltk:
+        >>> import nltk
+        >>> nltk.download()
+
+    This is a temporary fix to use the nltk library for your current terminal session on MacOS. If you want to make these changes permanent, you must add the commands from step 2 to your .bash_profile.
+
+Finally, after running `git submodule init`, you should be able to run the program using `python GUI.py`.
```
Comment (kpothuganti) [inline macos-setup.md:51]: can add another section for other issues that may occur (issues with pyaudio, tkinter, etc.) and include the commands that were used to fix them

File: macos-setup.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
+## Developer's Guide for Setting Up MacOS Environment
+
+Ensure you have the following prerequisites downloaded:
+
+* `Java` --> [Download Here](https://www.oracle.com/java/technologies/downloads/)
+* `MySQL` --> [Download Here](https://dev.mysql.com/downloads/mysql/)
+
+Once those are downloaded and you have cloned the Git repository, you will want to go into the SpeechTranscription directory and create/activate a Python virtual environment:
+
+* `python3 -m venv venv` to create the environment.
```
Comment (kpothuganti) [inline macos-setup.md:10]: can add a comment that the specific version of python might need to be included in this command

**Review by kpothuganti — CHANGES_REQUESTED** [review]
can add some more info regarding other issues that can occur during initial setup as well as mentioning that the python version used for the virtual environment may need to be specified

**Comment by kate-holdener** [comment]
These instructions are very useful!
I suggest putting all the developer specific information into a file called DEVELOPER_GUIDE.md (that's a standard file name used for developer instructions in open source repos). In that file, you can include the general developer info and the OS specific info. I recommend putting all instructions in the order that they should be followed, to prevent developers from running into the problems you ran into.

File: .gitignore [hunk 5]
```diff
@@ -6,4 +6,5 @@
 ffmpeg
 .env
 diarization.txt
-user_settings.txt
\ No newline at end of file
+user_settings.txt
+venv/
```
Comment (sagar3181) [inline .gitignore:10]: Good Code: code line 10
Good addition with venv/ in .gitignore.

File: README.md (hunk 2 above)
Comment (sagar3181) [reply README.md:21]: Consider expanding the macos-setup.md to include all OS setups or adding separate sections for Windows. This will make it easier for all developers to follow.

File: macos-setup.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+* `source venv/bin/activate` to activate.
```
Comment (sagar3181) [inline macos-setup.md:11]: mention that the application works best with Python 3.10.x to Python 3.11.x.

File: macos-setup.md (hunk 3 above)
Comment (sagar3181) [inline macos-setup.md:51]: Please confirm if "git submodule init" is sufficient on its own to initialize the submodules, or if we should also include "git submodule update" for a complete setup. This ensures the submodules are up-to-date after initialization. 2 steps; 1. "git submodule init" &  2. "git submodule update"

**Review by sagar3181 — CHANGES_REQUESTED** [review]
The PR is a solid step in improving the setup experience for macOS developers. the macos-setup.md file is detailed and covers the important steps, but there’s always a room to make it even more inclusive and robust by addressing the feedback mentioned

**Review by sagar3181 — APPROVED** [review]
Looks good for merge!
